app/src/main/java/com/kms/cura/view/activity/SearchActivity.java

```diff
@@ -146,5 +218,23 @@ public int compare(DoctorUserEntity u1, DoctorUserEntity u2) {
         }
     }
 
+    public void registerEvent() {
+        EventBroker.getInstance().register(this, EventConstant.FILTER_APPLIED);
+    }
+
+    @Override
+    public void handleEvent(String event, Object data) {
+        switch (event) {
+            case EventConstant.FILTER_APPLIED:
```

Comment:
If the dialog is modal then should not use event to handle it. Instead wait for dialog to be closed to update filter results.